### ISSUE TITLE
Model Release Discussions

### ISSUE THREAD

**Issue by lllyasviel** [lllyasviel/style2paints#205]
This issue is for the discussions of the release of Style2Paints V5 large model.

**Comment by xXChampionsXx** [comment]
This preview is a lot about image generation, but what about the line art coloring. Is there any preview for that... is it even still there in the first place?

**Comment by Ihateyoudattebayo** [comment]
So SEPA is an image diffusion model but V4 remaster will be a better version of V45?

**Comment by marcussacana** [comment]
First, what is the hardware minimal requeriments for run the IA?
Currently SableDiffusion projects allow set lowvram that can run in 4GB VRAM GPU even if slow, but what to expect from s2p v5?

**Comment by ShinkoNet** [comment]
Here's my opinion as an artist:
I think that this large model should be released as a seperate program to style2paints. The functionality is too different.
I liked style2paints v4 as a pipeline in the colouring process of a finished lineart. The program did not alter the lineart in any way, and it allowed me to pick which specific colours I wanted in each area. With this diffusion model, although the outputs are impressive, there is way less control over it in general. I also cannot do the lighting for myself, since the colouring, shading and lighting steps of v4 have become a single uncontrollable step in v5.
This large model does much better with loose sketch art, which gives it a different purpose more in line with brainstorming ideas, costumes and poses to use for an art piece, and works better placed in a different part of an art project pipeline than where I would place v4.

Anyway, I do agree that this model should be released for public use as its power is on the same level of Stable Diffusion's img2img mode, which is already public. The only advantage this model seems to give is that we don't have the need of colouring in a sketch yourself to see shaded variations, and that you can control how closely the model should adhere to the sketch in a better way than img2img can. But considering my above concerns, it should be renamed to a different software instead of an upgrade to an existing one.

**Comment by xXChampionsXx** [comment]
I completely agree with @ShinkoNet on this one. Image generation and lineart painting are 2 different pairs of shoes. The lineart was the main focus of the software and suddenly calling an image generation ai an upgrade makes no sense to me.

**Comment by lllyasviel** [comment]
About the previous coloring mode, see also an updated section "Workflow-Oriented Assistant". Note that the images in this section is not finished and for preview only.

**Comment by drbobo0** [comment]
thx lllyasviel and the S2P team for the update, i also agree with @ShinkoNet.
lllyasviel we need more info on it please keep us updated.

**Comment by lllyasviel** [comment]
Update:

After receiving an incredible number of feedbacks, we have decided to make a few extra technical preview rounds before we finalize V5.

A ‘legendary’ coloring mode will be put in a higher priority. This mode will produce results that preserve all details of input line drawings and sketches, while at the same time using an ‘inner’ generated diffusion reference image when making automatic color compositions. Please see also the newly added section [Workflow-Oriented Assistant] for a preview of this pipeline. Note that the images in this section is not finished and for preview only.

We underestimated the demand of a better ‘plain’ coloring mode. Fortunately, we noticed that right now, thanks to the supports and feedbacks from all of you.

Note that this will cause another delay on this project again. Nevertheless, and fortunately, we are much clearer than before about what to do and what should be done.

S2PR was and will always be a team focused on human-centric techniques, rather than AI-centric techniques.

Best,
S2PR

**Comment by lllyasviel** [comment]
If you have more considerations *other than* the ‘plain’ coloring mode, please also let us know by mentioning here.

**Comment by Ihateyoudattebayo** [comment]
> If you have more considerations _other than_ the ‘plain’ coloring mode, please also let us know by mentioning here.

I know this would be hard to integrate, but a mode similar to PaintingLight as well which can add lighting and textures.

**Comment by Ihateyoudattebayo** [comment]
> Update:
> 
> After receiving an incredible number of feedbacks, we have decided to make a few extra technical preview rounds before we finalize V5.
> 
> A ‘legendary’ coloring mode will be put in a higher priority. This mode will produce results that preserve all details of input line drawings and sketches, while at the same time using an ‘inner’ generated diffusion reference image when making automatic color compositions. Please see also the newly added section [Workflow-Oriented Assistant] for a preview of this pipeline. Note that the images in this section is not finished and for preview only.
> 
> We underestimated the demand of a better ‘plain’ coloring mode. Fortunately, we noticed that right now, thanks to the supports and feedbacks from all of you.
> 
> Note that the diffusion method can be used in many ways to also improve other components because it can generate automatic reference images for almost every pipeline.
> 
> Note that this will cause another delay on this project again. Nevertheless, and fortunately, we are much clearer than before about what to do and what should be done.
> 
> S2PR was and will always be a team focused on human-centric techniques, rather than AI-centric techniques.
> 
> Best, S2PR

I really apreciate the fact that S2PR is very open and uses the communities feedback. Also, what happened to SEPA UI?

**Comment by lllyasviel** [comment]
SEPA UI is also under heated disscussion of S2PR. Some people nowadays prefer Gradio more than a traditional desktop APP.

**Comment by Ihateyoudattebayo** [comment]
> SEPA UI is also under heated disscussion of S2PR. Some people nowadays prefer Gradio more than a traditional desktop APP.

In my opinion, a desktop app feels a lot more high quality and studio level, than a gradio web link.

**Comment by ShinkoNet** [comment]
Thank you for listening to feedback. I look forward to the colouring mode improvements.
I understand that the project hasn't shifted focus (looking at https://github.com/lllyasviel/style2paints/tree/master/V3#next-step ) in regard to improving images from rough sketches, so there are no problems with the diffusion model being added into the program. As long as the assisted colouring mode is still preserved.

Regarding the UI for SEPA, I do think that it should be kept to a standalone program instead of gradio. Gradio's library and functionality for image manipulation is quite limited. The web UI for hlky's stable diffusion fork had troubles with gradio when implementing img2img manipulation, for instance. It would be the best if we had enough visual tools to do a sketch, layering it, then lineart, then colour/lighting layer, all inside the same program. And gradio won't be good for that.

**Comment by TheZoroark007** [comment]
> Update:
> 
> After receiving an incredible number of feedbacks, we have decided to make a few extra technical preview rounds before we finalize V5.
> 
> A ‘legendary’ coloring mode will be put in a higher priority. This mode will produce results that preserve all details of input line drawings and sketches, while at the same time using an ‘inner’ generated diffusion reference image when making automatic color compositions. Please see also the newly added section [Workflow-Oriented Assistant] for a preview of this pipeline. Note that the images in this section is not finished and for preview only.
> 
> We underestimated the demand of a better ‘plain’ coloring mode. Fortunately, we noticed that right now, thanks to the supports and feedbacks from all of you.
> 
> Note that the diffusion method can be used in many ways to also improve other components because it can generate automatic reference images for almost every pipeline.
> 
> Note that this will cause another delay on this project again. Nevertheless, and fortunately, we are much clearer than before about what to do and what should be done.
> 
> S2PR was and will always be a team focused on human-centric techniques, rather than AI-centric techniques.
> 
> Best, S2PR

I would have a suggestion, but I'm not sure it really fits into SEPA: 
Would a mode that allows lineart to be automatically shaded like manga/doujin instead of color be possible ?

**Comment by Ihateyoudattebayo** [comment]
> 

Didnt they say that the idea was to use the diffusion generated img and feed it into the LEDGENDARY COLOURING MODE?

**Comment by bropines** [comment]
I did not see when this neural network model will be released. It is also possible for people who do not have a PC or a powerful PC to make a collab which will output the local server via bore.pub on the network. You can use https://github.com/carefree0910/carefree-creator as a webUI.

**Comment by bropines** [comment]
Also, why not use github discussions?

**Comment by lllyasviel** [comment]
Alice and Dorothy methods released. Future threads please comment based on the newer versions.

**Comment by lllyasviel** [comment]
> I did not see when this neural network model will be released. It is also possible for people who do not have a PC or a powerful PC to make a collab which will output the local server via bore.pub on the network. You can use https://github.com/carefree0910/carefree-creator as a webUI.

This is a great project but it is runned by a company and will add to the commercial factors that we want to avoid.

**Comment by p1atdev** [comment]
(I use DeppL because I am not very good at English.)

Style2Paints V5 seems to be using Anything V3.0, which is not desirable since Anything V3.0 uses a model leaked from NovelAI.
It is fine for those who are willing to use models that contain leaked ones, but it seems to me that it is better to avoid being a bad point for a good project like Style2Paints.
Is it possible instead to use a model without problems such as WaifuDiffusion?

**Comment by lllyasviel** [comment]
> (I use DeppL because I am not very good at English.)
> 
> Style2Paints V5 seems to be using Anything V3.0, which is not desirable since Anything V3.0 uses a model leaked from NovelAI. It is fine for those who are willing to use models that contain leaked ones, but it seems to me that it is better to avoid being a bad point for a good project like Style2Paints. Is it possible instead to use a model without problems such as WaifuDiffusion?

We cited Anything V3 because and only because we admire that model very much, not because that model is indispensible. Removing some dependency in model initialization is not difficult and seems to have nearly no influence on the final model. We do not even need WaifuDiffusion, alternatives can even be standard SD, since it is only training initilizations.

**Comment by ajundo** [comment]
Great Job, thanks for sharing very much.

I think the model works in a very similar way as https://huggingface.co/stabilityai/stable-diffusion-2-depth, which takes an extra depth map as conditional input. The depth model is finetuned from original SD without complex model modification(just adding an extra channel). 

> [1] The extra parameters are mainly in the encoding layers of the input sketch.

I see you changed the model structure a lot to handle to sketch input. Could you briefly introduce the benefits?

**Comment by lllyasviel** [comment]
> Great Job, thanks for sharing very much.
> 
> I think the model works in a very similar way as https://huggingface.co/stabilityai/stable-diffusion-2-depth, which takes an extra depth map as conditional input. The depth model is finetuned from original SD without complex model modification(just adding an extra channel).
> 
> > [1] The extra parameters are mainly in the encoding layers of the input sketch.
> 
> I see you changed the model structure a lot to handle to sketch input. Could you briefly introduce the benefits?

sketch has many sharp and sparse lines that tends to destroy the SD model in a direct fine tuning. Although SD is powerful, it is still a U-net: the skip-connection of unets will simply overfit to make unwanted edges to minimize the learning loss. When those 64*64 layers are overfit, all 8*8, 16*16, 32*32 layers will not learn anything. We need extra encoders to handle that.

Note that depth maps are dense-featured images and do not have this problem, and this is perhaps why stability use depth rather than image edges. 

If you directly finetune a SD model to learn edges, you will get something similar to this:

![image](https://user-images.githubusercontent.com/19834515/210730170-76a86a11-d911-4f09-b0c0-014076dfc048.png)


it is also in https://arxiv.org/abs/2012.09841

**Comment by Eric07110904** [comment]
Awesome works!!!
How much VRAM do I need for training or fine-tuning this model? 
I want to know the minimal requeriment of GPU.

**Comment by ajundo** [comment]
I see, you use a special encoder to encode the sketch because sketch has complicated semantic information. The information is not needed to be encode for a depth map feature. 
So if I understand correctly, the way of feed conditional sketch is similar to https://arxiv.org/pdf/1903.07291.pdf?
Thanks for your explanation.

**Comment by lllyasviel** [comment]
> I see, you use a special encoder to encode the sketch because sketch has complicated semantic information. The information is not needed to be encode for a depth map feature. So if I understand correctly, the way of feed conditional sketch is similar to https://arxiv.org/pdf/1903.07291.pdf? Thanks for your explanation.

We will release tech report soon. Just stay tuned!

**Comment by lllyasviel** [comment]
Update:

We removed Anything V3 from README. This is not because S2PR has taken any action. This is because some related people from Anything V3 contacted us and they do not want to claim the authority of that model since A3 is also a very wild mixture of lots of models and breambooths.

**Comment by Ihateyoudattebayo** [comment]
> Update:
> 
> We removed Anything V3 from README. This is not because S2PR has taken any action. This is because some related people from Anything V3 contacted us and they do not want to claim the authority of that model since A3 is also a very wild mixture of lots of models and dreambooths.

When is the colour scrible section going to be made?

**Comment by yakagami** [comment]
@lllyasviel I remember you said ML was a dead end for line art colorization, what changed? As for gradio vs sepaUI, gradio is still very limited. Manual corrections will always still be needed. For example see the hands of any of the examples- they are all inccorect. I think SD may be good for something like consistency between manga pannels and for following a scheme, but the origional S2P functionality was already very good and should not be removed in my opinion.

**Comment by SwayStar123** [comment]
This is amazing.

Will there be model distillation done to reduce inference times?

Is there any work being done on lineart -> lineart? Lineart would be easier to edit and would prove to be a good middle step before going straight to colorization (for the alice method, lineart -> lineart would provide greater control whilst still requiring much less detailed original input)

**Comment by Ihateyoudattebayo** [comment]
https://mobile.twitter.com/lvminzhang/status/1392143022221975554
 @lllyasviel 
What happened to this version of SEPA?
Also, what is the node system for?

**Comment by AltoCrat** [comment]
Hi, could I enquire about the acquisition of the dataset? 
"The training data comes from two domain: 50% are [Gwern](https://www.gwern.net/)’s Danbooru dataset (washed by many metrics), and 50% are research materials from Style2Paints Research (collected in 7 years from 2016 to 2023)."
I understand that the first dataset from Danbooru comes from Gwern, who appears to have permission to capture the data from the site itself for ML purposes. But I am curious about how the dataset for the research materials from Style2Paints Research was obtained, as I am interested in using the program but would like to avoid using data that could have been obtained without permission.
On another note, how possible would it be to train the model using my own drawings as data, on that end, would one really need as much as 1-7M pieces of unique data to train it, or could it be possible to train it with less? And what are the hardware limitations if so?
Any information is appreciated, as I am not very confident in my knowledge of AI models.

**Comment by lllyasviel** [comment]
Because the previous thread is out-of-date and many new previews are released, new thread starts [here](https://github.com/lllyasviel/style2paints/issues/207).
